solvebio/resource/dataset.py

```diff
@@ -333,27 +333,40 @@ def migrate(self, target, follow=True, **kwargs):
 
         return migration
 
-    def activity(self, follow=False):
+    def activity(self, follow=False, limit=1,
+                 wait_for_secs=Task.SLEEP_WAIT_DEFAULT):
+        """Get a list of active Tasks that have a target object of
+        the dataset. Active tasks are in the running, queued or pending state.
+
+        Defaults to limit=1 for performance. Increase this value in order
+        to return more tasks as output.
+        """
         statuses = ['running', 'queued', 'pending']
-        activity = list(Task.all(target_object_id=self.id,
-                                 status=','.join(statuses),
-                                 client=self._client))
 
-        print("Found {0} active task(s)".format(len(activity)))
+        # NOTE: source_object_id is not being queried here
+        # and therefore active DatasetExports will not appear
+        # in activity
+        activity = Task.all(target_object_id=self.id,
+                            status=','.join(statuses),
+                            limit=limit,
+                            ordering='-updated_at',
+                            client=self._client)
+
+        print("Found {0} active task(s)".format(activity.total))
         if not follow:
-            return activity
+            return list(activity)
 
         while True:
```

Comment:
To remove the recursion here, we could just move the `activity = Task.all()` to the top of the while loop, followed by `if not follow: return list(activity)`